src/core/quantum_circuit.jl

```diff
@@ -19,26 +19,30 @@ q[2]:
 """
 Base.@kwdef struct QuantumCircuit
     qubit_count::Int
-    gates::Vector{Gate} = Vector{Gate}([])
+    instructions::Vector{AbstractInstruction} = Vector{AbstractInstruction}([])
 
     function QuantumCircuit(
         qubit_count::Int,
-        gates::Vector{GateType},
-    ) where {GateType<:Gate}
+        instructions::Vector{InstructionType},
+    ) where {InstructionType<:AbstractInstruction}
         @assert qubit_count > 0 (
             "$(:QuantumCircuit) constructor requires qubit_count>0. Received: $qubit_count"
         )
 
         circuit = new(qubit_count, [])
-        foreach(gate -> ensure_gate_is_in_circuit(circuit, gate), gates)
-        append!(circuit.gates, gates)
+        foreach(instr -> ensure_instruction_is_in_circuit(circuit, instr), instructions)
+        append!(circuit.instructions, instructions)
         return circuit
     end
 end
 
+function Base.isequal(c0::QuantumCircuit, c1::QuantumCircuit)::Bool
```

Comment:
when I tried to `compare_circuit()` that contained `Readouts`, it would fail because we can simulate `Readouts`. This is to check that having readouts in transpiled circuits leaves the circuit in an equivalent state.